data/core/units/monsters/Fungoid_Body.cfg

```diff
@@ -0,0 +1,275 @@
+        arcane=200
+    [/resistance]
+    hitpoints=180
+    movement_type=smallfoot
+    movement=0
+    experience=150
+    {AMLA_DEFAULT}
+    level=3
+    alignment=chaotic
+    advances_to=null
+    cost=403
+    usage=mixed fighter
+    description= _ "This thing is impossible to describe, no one has seen anything like it before."
+    {DEFENSE_ANIM_FILTERED ("units/monsters/core-compact.png") ("units/monsters/core-compact.png") ("wail.wav") (direction=n,ne,se,s)}
+    {DEFENSE_ANIM_FILTERED ("units/monsters/core-compact.png~FL(horiz)") ("units/monsters/core-compact.png~FL(horiz)") ("wail.wav") (direction=nw,sw)}
+
+    [abilities]
+        {ABILITY_REGENERATES}
+        [dummy]
+            id=miasma
+            name= _ "miasma"
+            description=_ "This unit passively harms those not of its side at a distance and rate equal to its level."
+        [/dummy]
+    [/abilities]
+    # Miasma ability.
+    [event]
+        id=Fungoid Miasma
+        name=turn end
+        first_time_only=no
+        [store_unit]
+            [filter]
+                type="Fungoid Spore","Fungoid Stalk","Fungoid Body"
+            [/filter]
+            variable=fungoids
+        [/store_unit]
+        [foreach]
+            array=fungoids
+            variable=fungoid
+            [do]
+                [harm_unit]
+                    [filter]
+                        [filter_location]
+                            [filter]
+                                id=$fungoid.id
+                            [/filter]
+                            radius=$fungoid.level
+                        [/filter_location]
+                        [not]
+                            side=$fungoid.side
+                        [/not]
+                    [/filter]
+                    [filter_second]
+                        id=$fungoid.id
+                    [/filter_second]
+                    amount=$fungoid.level
+                    damage_type=cold
+                    fire_event=yes
+                    animate=yes
+                [/harm_unit]
+            [/do]
+        [/foreach]
+    [/event]
+
+    # A fruiting body feeds whenever its spores and stalks do.
+    [event]
+        name=attacker hits
+        first_time_only=no
+        [filter]
+            type=Fungoid Spore,Fungoid Stalk
+        [/filter]
+        [heal_unit]
+            [filter]
+                type=Fungoid Body
+            [/filter]
+            amount=$damage_inflicted
+            animate=yes
+        [/heal_unit]
+    [/event]
+    [event]
+        name=defender hits
+        first_time_only=no
+        [filter_second]
+            type=Fungoid Spore,Fungoid Stalk
+        [/filter_second]
+        [heal_unit]
+            [filter]
+                type=Fungoid Body
+            [/filter]
+            amount=$damage_inflicted
+            animate=yes
+        [/heal_unit]
+    [/event]
+
+    # Every turn there is a chance a fruiting body will produce a new spore unit.
```

Comment:
I think there should be a `[dummy]` ability documenting this fact.

Reply:
Yeah that is a good point. If this mechanic remains directly part of the `unit_type` then that will should be the way to handle it. In this present form it might need a very long description though, to cover all the mechanics involved.

Reply:
That makes sense.